src/agentscope/utils/mixture_of_agent.py

```diff
@@ -0,0 +1,167 @@
+        )
+        for i, ref in enumerate(self.references, start=0):
+            messages.append(
+                Msg(
+                    role="user",
+                    content=ref,
+                    name=f"Model_{i}",
+                ),
+            )
+        aggre_format_msg = aggre_model.format(messages)
+        aggre_res = aggre_model(aggre_format_msg)
+        return aggre_res.text
+
+    def reply(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> Msg:
+        """
+        Get model response from messages.
+        Is equivalent to calling a model with:
+            ```
+            format_msg = model.format(messages)
+            return model(format_msg)
+            ```
+
+        Args:
+            *args (`Union[Msg, Sequence[Msg]]`):
+                The messages to be sent to the model.
+        """
+
+        def _process_reference(
+            i: int,
+            ref_model: ModelWrapperBase,
+            *args: Union[Msg, Sequence[Msg]],
+        ) -> Tuple[int, str]:
+            format_msg = ref_model.format(*args)
+            ref_model_res = ref_model(format_msg)
+            return i, ref_model_res.text
+
+        def _process_new_refs(
+            i: int,
+            ref_model: ModelWrapperBase,
+        ) -> Tuple[int, str]:
+            return i, self._get_res_with_aggregate_model(ref_model)
+
+        # get all the references
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = [
+                executor.submit(_process_reference, i, ref_model, *args)
+                for i, ref_model in enumerate(self.reference_models, start=0)
+            ]
+            for future in concurrent.futures.as_completed(futures):
+                i, result = future.result()
+                self.references[i] = result
+                if self.show_internal:
+                    print(f"Round {0}, Model_{i}: {result}")
+
+        for r in range(self.rounds):
+            if self.show_internal:
+                print("=" * 20)
+            new_refs = ["" for _ in range(len(self.reference_models))]
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                futures = [
+                    executor.submit(_process_new_refs, i, ref_model)
+                    for i, ref_model in enumerate(
+                        self.reference_models,
+                        start=0,
+                    )
+                ]
+                for future in concurrent.futures.as_completed(futures):
+                    i, result = future.result()
+                    new_refs[i] = result
+                    if self.show_internal:
+                        print(f"Round {r+1}, Model_{i}: {result}")
+            self.references = new_refs
+
+        final_res = self._get_res_with_aggregate_model(self.main_model)
+        return Msg(role="assistant", content=final_res, name="assistant")
```

Comment:
In current implementation, this module is much more like an agent rather than a module. For example, we should provide the `role` and `name` here, which is owned by an agent rather than a module.

Reply:
Changed to return str instead.